Adds README overview (#1)
http-wasm/http-wasm-abi: README.md

has a chain of middleware that allows it to customize or change requests and
responses. Middleware in Wasm can be included in this chain.

```goat
 .----------------------. 
|                        |
|  .------------------.  |
| | HTTP Processor     | |
|  '------+-----------'  | 
|         |  ^           |
|         v  |           |
|  .---------+--------.  |
| | Native Middleware  | |
|  '------+-----------'  | 
|         |  ^           |
|         v  |           |
|  .---------+--------.  |
| | Wasm Middleware    | |
| |                    | |
| | .-----.    .-----. | |
| || Host  |  | Guest || |
| ||       +->|       || |
| ||       |<-+       || |
| | '-----'    '-----' | |
|  '------+-----------'  | 
|         |  ^           |
|         v  |           |
|  .---------+--------.  |
| | Native Middleware  | |
|  '------------------'  | 
|                        |
 '----------------------' 
```

The Wasm Middleware *host* is written in native code, and compiled into the
application. For example, this code could use [wazero][2] as the WebAssembly
runtime, if the host was written in Go, or [V8][3], if C++.

> The Wasm Middleware *guest* can be written in any language that supports the
> ABI in use by the *host*. For example, the middleware could be programmed in C

Reviewer: ```suggestion
The Wasm Middleware *guest* can be written in any language that can be compiled
to Wasm. For example, the middleware could be programmed in C
```

I don't think there's any "support" a language provides for an ABI, it's the actual code that would use it.
